server/utils/canvas.py

```diff
@@ -0,0 +1,83 @@
+from flask import session, request, url_for
+from canvasapi import Canvas
+
+from server import app
+from server.models import Offering
+from server.utils.stub import get_dev_user, get_dev_course, get_dev_user_courses
+from server.utils.url import Redirect
+
+
+def _get_client(key=None):
+    if not key:
+        key = session.get('access_token', None)
+    if not key:
+        session['after_login'] = request.url
+        raise Redirect(url_for('login'))
+    return Canvas(app.config['CANVAS_SERVER_URL'], key)
+
+
+def is_mock_canvas():
+    return app.config['MOCK_CANVAS'] and \
+        app.config['FLASK_ENV'].lower() != 'production'
+
+
+def get_user(canvas_id, key=None):
+    if is_mock_canvas():
```

Comment:
Yeah seems like we might want a custom MockCanvasClient

Reply:
True, that's a way better design


I am also looking at mocking libraries to provide a more generalized solution. Will probably integrate this [library](https://github.com/getsentry/responses).